File: clients/javascript/src/models/publishOutMsg.ts
// this file is @generated





export interface PublishOutMsg {
    offset: number;
topic: string;
}

export const PublishOutMsgSerializer = {
    _fromJsonObject(object: any): PublishOutMsg {
        return {
            offset: object['offset'],
            topic: object['topic'],
            };
    },

    _toJsonObject(self: PublishOutMsg): any {
        return {
            'offset': self.offset,
            'topic': self.topic,
            };
    }
}

File: clients/rust/src/models/publish_out_msg.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct PublishOutMsg {
    pub offset: u64,

    pub topic: String,
}

impl PublishOutMsg {
    pub fn new(offset: u64, topic: String) -> Self {
        Self { offset, topic }
    }
}
